### PR TITLE
Use Pwm from HAL crate

### PR DESCRIPTION
**Pull request by qwandor**
I moved a bunch of the code into the `gd32f1x0-hal` crate in https://github.com/qwandor/gd32f1x0-hal/pull/7. This should be functionally identical.

The main thing to note is that I'm using a different PWM mode, which means that the duty cycles essentially mean the opposite of what they did before, and I've compensated by negating the power so that the direction is still the same.

TODO:
- [x] Test on dev board to make sure PWM waveforms look the same with and without this change.
- [x] Test on real board to make sure it still behaves the same with and without this change (e.g. that the directions and speeds are the same).
- [x] Test that break input works as expected.
- [x] Figure out how the latest HAL crate broke setting the buzzer frequency.

### CODE REVIEW

File: cross/hoverkite-firmware/src/motor.rs [hunk 1]
```diff
@@ -64,10 +74,71 @@ pub struct Motor {
     power: i16,
     /// The number of timer cycles since the motor power was last changed.
     smoothing_cycles: u32,
+    _emergency_off: PB12<Alternate<AF2>>,
```
Comment (alsuren) [inline cross/hoverkite-firmware/src/motor.rs:77]: comment about this being a placeholder and wanting to push this into Pwm at some point in the future?

File: cross/hoverkite-firmware/src/motor.rs [hunk 2]
```diff
@@ -64,10 +74,71 @@ pub struct Motor {
...
+    pwm.automatic_output_disable();
+
+    // Enable PWM output on all channels and complementary channels.
+    for channel in &channels {
+        pwm.enable(*channel);
+    }
+
+    // Enable timer interrupt
+    // TODO: Set priority?
+    pwm.listen(Event::Update);
```
Comment (alsuren) [inline cross/hoverkite-firmware/src/motor.rs:126]: does this todo have a ticket?

**Review by alsuren — APPROVED** [review]

File: cross/hoverkite-firmware/src/motor.rs (hunk 2 above)
Comment (qwandor) [reply cross/hoverkite-firmware/src/motor.rs:126]: Removed it.

File: cross/hoverkite-firmware/src/motor.rs (hunk 1 above)
Comment (qwandor) [reply cross/hoverkite-firmware/src/motor.rs:79]: Done.

**Comment by qwandor** [comment]
Fixed the buzzer bug in https://github.com/qwandor/gd32-rs/pull/18.

**Pull request merged**